scripts/ci/junit2slack/main.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/GoogleCloudPlatform/testgrid/metadata/junit"
+	"github.com/slack-go/slack"
+	"log"
+	"os"
+)
+
+func main() {
+	// We're only logging errors currently. If we log more, we should probably switch logging libraries
+	log.SetOutput(os.Stderr)
+
+	var slackAttachments []slack.Attachment
+	var junitFiles []*junit.Suites
+
+	// We should accept all file names at once since we're using `go run` to run this program. No need to recompile
+	// for each file we want to parse
+	for _, fileName := range os.Args[1:] {
+		if _, err := os.Stat(fileName); err == nil {
+			data, err := os.ReadFile(fileName)
+			if err != nil {
+				log.Printf("error while reading %s: %s", fileName, err)
+				continue
+			}
+
+			junitSuites, err := junit.Parse(data)
+			if err != nil {
+				log.Printf("error while parsing junit suites in %s: %s", fileName, err)
+				continue
+			}
+			junitFiles = append(junitFiles, junitSuites)
+
+		} else if errors.Is(err, os.ErrNotExist) {
+			log.Printf("%s doesn't exist: %s", fileName, err)
+		} else {
+			log.Printf("error while trying to find %s: %s", fileName, err)
+		}
+	}
+
+	slackMsg := convertJunitToSlack(junitFiles)
+	if slackMsg == nil {
+		log.Printf("warning: no slack message set")
+		return
+	} else {
+		slackAttachments = append(slackAttachments)
+	}
+
+	b, err := json.Marshal(slackMsg)
+	if err != nil {
+		log.Printf("error while marshaling Slack message to json: %s", err)
+	}
+	fmt.Println(string(b))
+}
+
+func convertJunitToSlack(junitFiles []*junit.Suites) []slack.Attachment {
+	var failedTestsBlocks []slack.Block
+	var attachments []slack.Attachment
+
+	for _, suites := range junitFiles {
+		for _, suite := range suites.Suites {
+			// We currently only care about failures
+			if suite.Failures == 0 {
+				continue
+			}
+
+			for _, result := range suite.Results {
+				// We currently only care about failures
+				if result.Failure == nil {
+					continue
+				}
+
+				var title string
+				if result.ClassName == "" {
+					title = result.Name
+				} else {
+					title = fmt.Sprintf("%s: %s", result.ClassName, result.Name)
+				}
+
+				titleTextBlock := slack.NewTextBlockObject("plain_text", title, false, false)
+				titleSectionBlock := slack.NewSectionBlock(titleTextBlock, nil, nil)
+				failedTestsBlocks = append(failedTestsBlocks, titleSectionBlock)
+
+				failureMessage := result.Failure.Message
+
+				// If there's no failure message, we'll use a different message (this shouldn't be the usual case)
+				if len(failureMessage) <= 0 {
```

Comment:
you can just check `== 0`

Reply:
Generally I would prefer `failureMessage == ""` in Go. This is the most direct and has no performance disadvantages over `len(...) == 0`